metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -121,6 +125,207 @@ final class BloopServers(
+    )
+  }
+
+  private def getBloopGlobalJsonLastModifiedByMetalsTime(
+      bloopCreatedByMetalsFilePath: AbsolutePath
+  ): Long = Try {
+    bloopCreatedByMetalsFilePath.readText.toLong
+  }.getOrElse(0)
+
+  private def processUserPreferenceForBloopJvmProperties(
+      messageActionItem: MessageActionItem,
+      bloopGlobalJsonFilePath: AbsolutePath,
+      bloopCreatedByMetalsFilePath: AbsolutePath,
+      requestedBloopJvmProperties: List[String],
+      maybeJavaHome: Option[String],
+      reconnect: () => Future[BuildChange]
+  ): Future[Unit] = {
+    messageActionItem match {
+
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.applyAndRestart =>
+        writeJVMPropertiesToBloopGlobalJsonFile(
+          bloopGlobalJsonFilePath,
+          bloopCreatedByMetalsFilePath,
+          requestedBloopJvmProperties,
+          maybeJavaHome
+        ) match {
+          case Failure(exception) => Future.failed(exception)
+          case Success(_) =>
+            shutdownServer()
+            reconnect().ignoreValue
+        }
+
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.openGlobalJsonFile =>
+        val position = new Position(0, 0)
+        val range = new org.eclipse.lsp4j.Range(position, position)
+        val command = ClientCommands.GotoLocation.toExecuteCommandParams(
+          ClientCommands.WindowLocation(
+            bloopGlobalJsonFilePath.toURI.toString,
+            range
+          )
+        )
+        Future.successful(languageClient.metalsExecuteClientCommand(command))
+
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.useGlobalFile =>
+        Future.successful(())
+    }
+  }
+
+  private def updateBloopGlobalJsonFileThenRestart(
+      bloopGlobalJsonFilePath: AbsolutePath,
+      bloopCreatedByMetalsFilePath: AbsolutePath,
+      requestedBloopJvmProperties: List[String],
+      maybeJavaHome: Option[String],
+      reconnect: () => Future[BuildChange]
+  ): Future[Unit] = {
+    writeJVMPropertiesToBloopGlobalJsonFile(
+      bloopGlobalJsonFilePath,
+      bloopCreatedByMetalsFilePath,
+      requestedBloopJvmProperties,
+      maybeJavaHome
+    ) match {
+      case Failure(exception) => Future.failed(exception)
+      case Success(_) =>
+        languageClient
+          .showMessageRequest(
+            Messages.BloopJvmPropertiesChange.params()
+          )
+          .asScala
+          .flatMap {
+            case messageActionItem
+                if messageActionItem == Messages.BloopJvmPropertiesChange.reconnect =>
+              shutdownServer()
+              reconnect().ignoreValue
+            case _ =>
+              Future.successful(())
+          }
+    }
+
+  }
+
+  private def getBloopGlobalJsonFilePath(): Option[AbsolutePath] =
+    Try {
+      AbsolutePath(
+        Paths
+          .get(System.getProperty("user.home"))
+          .resolve(".bloop/bloop.json")
```

Comment:
We check for `.bloop/bloop.json`, but what about users that are utilizing the `.bloop/.jvmopts` file? For example that's what I was using locally to set my Bloop jvm opts, and now I have both that file and a newly created `.bloop/bloop.json` file.

Reply:
ah, thank you very much for providing this extra information. This can definitely cause problems. And is the contents of the `.bloop/.jvmopts` file supposed to be the same as the `.bloop/bloop.json` one?
What are your suggestions for the logic of resolving this when the `.jvmopts` file is present?

Reply:
Well I'm not sure actually. @tgodzik is it not recommend with Bloop to use the `.bloop/.jvmopts` file? It might just be this is legacy and I shouldn't be doing it.

Reply:
I think the best option would be to follow similar thing as for an existing `bloop.json` but the user decides we should remove `.jvmopts` file and otherwise suggest they migrate to `bloop.json` themselves

`.jvmopts` format is just a list of options, it's not a json file at all.

Reply:
https://github.com/scalacenter/bloop/pull/1707

Reply:
Yup, perfect 👍🏼